lib/libc/semaphore/sem_init.c

```diff
@@ -60,11 +60,41 @@
 #include <limits.h>
 #include <semaphore.h>
 #include <errno.h>
+#if defined(CONFIG_BINMGR_RECOVERY) && defined(__KERNEL__)
+#include <queue.h>
+#include <tinyara/arch.h>
+#endif
 
 #ifdef CONFIG_SEMAPHORE_HISTORY
 #include <tinyara/debug/sysdbg.h>
 #endif
 
+#if defined(CONFIG_BINMGR_RECOVERY) && defined(__KERNEL__)
+sq_queue_t g_sem_list;
+
+static void sem_register(FAR sem_t *sem)
+{
+	sem_t *sem_ptr;
+	irqstate_t flags;
+
+	flags = irqsave();
+	if (!sq_empty(&g_sem_list)) {
```

Comment:
I think this if condition is not needed, because if g_sem_list is NULL, then below sem_ptr will be NULL.